pkg/network/setup/netstat.go

```diff
@@ -114,45 +114,64 @@ func (c *NetStat) UpdateStatus(vmi *v1.VirtualMachineInstance, domain *api.Domai
 	interfacesStatus = ifacesStatusFromGuestAgent(interfacesStatus, domain.Status.Interfaces)
 
 	primaryNetwork := netvmispec.LookupPodNetwork(vmi.Spec.Networks)
-	interfacesStatus = restorePodIfaceNameForPrimaryIface(interfacesStatus, primaryNetwork, vmi.Status.Interfaces)
+	if primaryNetwork != nil {
```

Comment:
> The fact we use previous status to calculate the new status is problematic because we make the status statefull, it seems to diverge from the common convention and best practices.
I am aware this PR does not introduce this pattern, but I think it should not extend it at least.

The VMI is owned by (at least) the VMI controller and virt-handler.
When the infosource field was introduced by https://github.com/kubevirt/kubevirt/pull/6793, it caused the VMI.Status.Interfaces field to be owned by both controllers (each owning several fields).
So, they both need to keep each-other's data when calculating their own status.

> It seems update-status should retain pod-interface-name, can you tell where is it being overridden?

Each controller builds its own info and preserves the other's data (this PR extends this pattern).

Reply:
Looking at this again, it seems I got it wrong, its not actually looks at previous iteration status, it "restores" the podIfaceName attribute that overriden by some other parts of UpdateStatus.

The "restoring" approach is confusing, and it make the code not consistent (build a iface list, override attributes, "restore" attributes..)  its hard to follow.
As mentioned above the `ifacesStatusFromMultus` seem better fit for populating the podIfaceName in this context.

Reply:
`ifacesStatusFromMultus` does basically the same thing.

Reply:
> As mentioned above the ifacesStatusFromMultus seem better fit for populating the podIfaceName in this context.

`ifacesStatusFromMultus` basically does the same thing.

Reply:
Sorry I dont understand what you mean by that, `ifacesStatusFromMultus` doesnt touch `podIfaceName`.
I am asking for changing `ifacesStatusFromMultus` to populate the `podIfaceName`, if possible.
That way we can avoid the restore approach I find confusing.

Reply:
> The update-status logic is already complex and had to follow due the fact interface status is composed from multiple sources. This change extends the interface status composition from multus source. But it seem to contradict with other sources (GA?), forcing the restore pattern.

There are several distinct phases:
1. Calculating the interface status from:
- Domain
- Host devices (SR-IOV)
- Pod cache
- Guest Agent
2. Restoration of primary interface status (if VMI controller had reported it)
3. Restoration of Multus status info source
4. Restoration of PodInterfaceNames

Steps 2-4 are restoring information set by VMI controller (multus infosource and PodInterfaceName).

> I find having the podIfaceName restore logic hard to follow in this context (which is already complex), I hope we can avoid it. How about populating the podIfaceName once, after population the iface status from other sources? (by `ifacesStatusFromMultus`).

This is what the code does.

Reply:
I am confused by the "restore" wording in this function context, it make me think somthing was deleted and we need to restore it, at least for me. 
(I would call this operation add/update/populate)
Following offline discussion, I realized there is no actual restoration and the current impl. functions.

I do see some complexity added I wish we could avoid, specifically with (2) & (4), I think we can combine them.
For example: set the primary iface `podIfaceName` with default value, and then call the unified `podIfaceName` population function, having it override the default if there is relevant data from Multus.

I am not sure how much time that will take. Not a must for this PR.